muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/actions/main-function/users/index.ts

```diff
@@ -70,30 +83,53 @@ let applyStudent: ApplyStudentTriggerType = function applyStudent (data) {
   }
 }
 
-const selectStaffmemberEndPoint:  ApplyStaffmemberTriggerType = function studentEndPoint (data) {
-  if(data.mode  == "CREATE" ) {
-    return mApi().user.staffMembers.create(data.staffmember);
-  } else {
-    return mApi().user.staffMembers.put(data.staffmember);
+let updateStudent: UpdateStudentTriggerType = function updateStudent(data) {
+  return async (dispatch: (arg: AnyActionType) => any, getState: () => StateType) => {
+    try {
+      await promisify(mApi().user.students.basicInfo.update(data.student.identifier, data.student), 'callback')().then(() => {
+
+        mApi().organizationUserManagement.students.cacheClear();
+
+        setTimeout(async () => {
```

Comment:
Because update : muikku --> pyramus ---> muikku --> and if it just refreshes, it's too fast and no changes in state. I made it 2 second just to be sure.